Strata/Languages/B3/DDMTransform/Translate.lean

```diff
@@ -0,0 +1,283 @@
+/-
+  Copyright Strata Contributors
+
+  SPDX-License-Identifier: Apache-2.0 OR MIT
+-/
+
+import Strata.DDM.AST
+import Strata.Languages.B3.DDMTransform.Parse
+import Strata.Languages.B3.Stmt
+
+---------------------------------------------------------------------
+namespace Strata
+
+/- Translating concrete syntax into abstract syntax -/
+
+open B3
+open Std (ToFormat Format format)
+
+---------------------------------------------------------------------
+
+/- Translation Monad -/
+
+structure TransState where
+  errors : Array String
+
+def TransM := StateM TransState
+  deriving Monad
+
+def TransM.run (m : TransM α) : (α × Array String) :=
+  let (v, s) := StateT.run m { errors := #[] }
+  (v, s.errors)
+
+def TransM.error [Inhabited α] (msg : String) : TransM α := do
+  fun s => ((), { errors := s.errors.push msg })
+  return panic msg
+
+---------------------------------------------------------------------
+
+def checkOp (op : Operation) (name : QualifiedIdent) (argc : Nat) :
```

Comment:
Are there issues with the #strata_gen implementation that preclude using it in this file?